# Generated by Django 4.2.23 on 2025-07-26 20:35

from django.db import migrations, models
import django.db.models.deletion
import uuid


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        ('mailing_campaigns_core', '0001_initial'),
    ]

    operations = [
        migrations.CreateModel(
            name='CampaignDelivery',
            fields=[
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('updated_at', models.DateTimeField(auto_now=True)),
                ('id', models.UUIDField(default=uuid.uuid4, editable=False, primary_key=True, serialize=False)),
                ('delivery_status', models.CharField(choices=[('pending', 'Pending'), ('processing', 'Processing'), ('completed', 'Completed'), ('failed', 'Failed'), ('cancelled', 'Cancelled')], default='pending', max_length=30)),
                ('sending_server', models.CharField(blank=True, max_length=100)),
                ('sending_ip', models.GenericIPAddressField(blank=True, null=True)),
                ('total_to_send', models.IntegerField(default=0)),
                ('sent_count', models.IntegerField(default=0)),
                ('failed_count', models.IntegerField(default=0)),
                ('processing_rate', models.IntegerField(default=0)),
                ('estimated_completion', models.DateTimeField(blank=True, null=True)),
                ('started_at', models.DateTimeField(blank=True, null=True)),
                ('completed_at', models.DateTimeField(blank=True, null=True)),
                ('error_message', models.TextField(blank=True)),
                ('campaign', models.OneToOneField(on_delete=django.db.models.deletion.CASCADE, to='mailing_campaigns_core.campaign')),
            ],
            options={
                'abstract': False,
            },
        ),
    ]
